spec/lib/canvas/validators/schema_attributes/experience_slot_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+describe Canvas::Validator::SchemaAttribute::ExperienceSlot do
+  subject(:validator) { described_class.new(attribute) }
+
+  describe "#validate" do
+    describe "validating an optional 'default' key" do
+      let(:attribute) {
+        {
+          "name" => "my_slot",
+          "type" => "experience_slot",
+          "default" => default_value
+        }
+      }
+
+      context "when unknown default value is provided" do
+        let(:default_value) { "FAIL" }
+
+        it "adds an error and fails the validation" do
+          expect(validator.validate).to eq(false)
+          expect(validator.errors).to include(
+            "\"default\" for experience_slot variables must be one of: next_upcoming"
+          )
+        end
+      end
+
+      context "when `next_upcoming` default value is provided" do
+        let(:default_value) { "Next_upcoming" }
+
+        it "passes the validation" do
+          expect(validator.validate).to eq(true)
+        end
+      end
+
+      context "when `default` is an array of invalid options" do
```

Comment:
Does this context need to be "when `default` is an array containing an invalid option"? Just wondering as one of the options is valid and the other isn't and this seems to suggest all of them are.

Reply:
Nice yes, that's more specific. I took this wording (and spec structure) directly from the ones for experience_date so I've updated the wording there too 👍